sdk/azcore/doc.go

```diff
@@ -5,160 +5,125 @@
-a reference to the next Policy object in the linked list. The New method should create its corresponding Policy object
-passing it the PolicyNode and any other configuration/settings fields appropriate for the specific Policy object.
+         // forward Request to next Policy & get Response/error
+         resp, err := req.Next()
 
-Here is a template for how to define a pipeline.Policy object:
+         // TODO: mutate/process Response/error here
 
-   // NOTE: Once created & initialized, Factory objects should be goroutine-safe (ex: immutable);
-   // this allows reuse (efficient use of memory) and makes these objects usable by multiple goroutines concurrently.
-   type myPolicyFactory struct {
-      // TODO: Add any configuration/setting fields if desired...
+         // return Response/error to previous Policy
+         return resp, err
+	   })
    }
 
-   func (f *myPolicyFactory) New(node pipeline.PolicyNode) Policy {
-      return &myPolicy{node: node} // TODO: Also initialize any configuration/setting fields here (if desired)...
+Template for implementing a stateful Policy:
+
+   type MyStatefulPolicy struct {
+      // TODO: add configuration/setting fields here...
    }
 
-Using your Factory and Policy objects via a Pipeline
+   func (p *MyStatefulPolicy) Do(req *azcore.Request) (resp *azcore.Response, err error) {
+         // TODO: mutate/process Request here
 
-To use the Factory and Policy objects, an application constructs a slice of Factory objects and passes
-this slice to the pipeline.NewPipeline function.
+         // forward Request to next Policy & get Response/error
+         resp, err := req.Next()
 
-   func NewPipeline(factories []pipeline.Factory, sender pipeline.HTTPSender) Pipeline
+         // TODO: mutate/process Response/error here
 
-This function also requires an object implementing the HTTPSender interface. For simple scenarios,
-passing nil for HTTPSender causes a standard Go http.Client object to be created and used to actually
-send the HTTP response over the network. For more advanced scenarios, you can pass your own HTTPSender
-object in. This allows sharing of http.Client objects or the use of custom-configured http.Client objects
-or other objects that can simulate the network requests for testing purposes.
+         // return Response/error to previous Policy
+         return resp, err
+   }
 
-Now that you have a pipeline.Pipeline object, you can create a pipeline.Request object (which is a simple
-wrapper around Go's standard http.Request object) and pass it to Pipeline's Do method along with passing a
-context.Context for cancelling the HTTP request (if desired).
+Implementing the Transport Interface
 
-   type Pipeline interface {
-      Do(ctx context.Context, methodFactory pipeline.Factory, request pipeline.Request) (pipeline.Response, error)
-   }
+The Transport interface is a specialized Policy.  Its responsibility is to send the HTTP request
```

Comment:
In C/C++ we have a TransportPolicy and its option is a Transport (which is not a policy). Our TransportPolicy calls the Transport and then reads the body (unless its a blob/file download). I like this approach and I think this is what NewPipline expect too. So, maybe the doc wording here just needs some cleanup with respect to terminology.

For example, you say "The Trasnport is always the last policy in the chain" but Transport is not a policy.

Reply:
Fixed